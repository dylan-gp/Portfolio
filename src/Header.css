.header {
  /* height: 65vw; */
  height: 100vh;
  overflow: hidden;
  background-color: rgba(67, 32, 160, .8);
  position: relative;
  z-index: 3;
}

.header-titles {
  margin: auto;
  /* height: 54.2vw; */
  height: 80vh;
  display: flex;
  flex-direction: column;
  justify-content: center;
}

.header-hello {
  /* font-size: 4vw; */
  font-size: 7vmin;
  font-weight: 100;
  color: black;
  display: flex;
  align-self: center;
   -webkit-text-fill-color: rgba(67, 32, 160, .8); /* Will override color (regardless of order) */
   -webkit-text-stroke-width: .3vmin;
   -webkit-text-stroke-color: black;
   text-shadow:
		-.2vmin -.2vmin 0 #000,
		.2vmin -.2vmin 0 #000,
		-.2vmin .2vmin 0 #000,
		.2vmin .2vmin 0 #000;
}
.arrow {
  height: 10vmin;
  position: absolute;
  top: 70%;
  left: 50%;
  -webkit-transition: 0.4s;
  -moz-transition: 0.4s;
  -o-transition: 0.4s;
  transition: 0.4s;
  opacity: 1;
  visibility: visible; 
}

.arrow-none {
  height: 10vmin;
  position: absolute;
  top: 70%;
  left: 50%;
  -webkit-transition: 0.4s;
  -moz-transition: 0.4s;
  -o-transition: 0.4s;
  transition: 0.4s;
  opacity: 0;
  visibility: hidden; 
}

.header-title {
  padding: 0;
  margin: .6vw auto;
  text-align: center;
  font-size: 8vmin;
  -webkit-transition: 0.8s;
  -moz-transition: 0.8s;
  -o-transition: 0.8s;
  transition: 0.8s;
  opacity: 1;
  display: flex;
  align-self: center;
  visibility: visible; 
  /* text-shadow: 2px 2px rgba(255, 255, 255, .6); */
}

.header-title-none {
  display: flex;
  align-self: center;
  -webkit-transition: 0.8s;
  -moz-transition: 0.8s;
  -o-transition: 0.8s;
  transition: 0.8s;
  font-size: 8vmin;
  opacity: 0;
  visibility: hidden; 
}

.header-links {
  height: 20vmin;
  display: flex;
  flex-direction: row;
  justify-content: center;
  align-items: center;
  align-content: center;
  position:relative;
  z-index:9999;
}
.header-icon {
  height: 9vmin;
  margin: auto .6vw;
  
}