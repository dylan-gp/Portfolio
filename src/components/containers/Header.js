import React from 'react';
import HeaderView from '../presentation/HeaderView';

export default class Header extends React.Component {
  state = {
    hide: true,
    arrow: false,
    hello: "",
    particle: true,
  }
  helloWorld = "Hello World";
  componentDidMount() {
    window.addEventListener('scroll', this.hideTitle);
    const element = document.getElementById('loader');
    const bounce1 = document.getElementById('bounce1');
    const bounce2 = document.getElementById('bounce2');
    const bounce3 = document.getElementById('bounce3');
    if(element){
      setTimeout(() => {
        setTimeout(() => element.classList.add('available'), 500);
        bounce1.classList.add('final-bounce');
        bounce2.classList.add('final-bounce');
        bounce3.classList.add('final-bounce');
        document.body.classList.remove('no-scroll');
        setTimeout(() => this.setState({ hide: false }), 2000);
        this.writeHello();
        setTimeout(() => this.interval = setInterval(() => this.arrowFlash(), 1200), 3500);
        setTimeout(() => {
          element.parentNode.removeChild(element);     
        }, 2000)
      }, 3000);
    }
  }
  componentWillUnmount() {
    window.removeEventListener('scroll', this.hideTitle);
    clearInterval(this.interval);
  }
  hideTitle = () => {
    window.scrollY > 0 ?
      !this.state.hide ? this.setState({ hide: true }) : null:
      this.state.hide ? this.setState({ hide: false }) : null;
    if (this.header.getBoundingClientRect().bottom <= 20 
      && this.state.hello === "Hello World") 
      this.setState({ hello: "Hello Again" });
    if (window.scrollY >= 700) clearInterval(this.interval);
  }

  arrowFlash = () =>
    this.setState({ arrow: !this.state.arrow });

  writeHello = () =>
    setTimeout(
      () => {
        const int = setInterval(
          () => {
            this.setState(
              {
                hello: `${this.state.hello}${this.helloWorld.slice(0, 1)}`
              }
            );
            this.helloWorld = this.helloWorld.slice(1);
            if (this.helloWorld.length === 0)
              clearInterval(int);
          },
          100
        );
      },
      2500
    );
  
  setHeaderRef = (element) =>
    this.header = element;

  render = () =>
    <HeaderView
      setHeaderRef={this.setHeaderRef}
      state={this.state}
    />;

}