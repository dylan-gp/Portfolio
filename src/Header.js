import React from 'react';
import Particles from 'react-particles-js';
import mypicture from './mypicture.jpg';
import linkedinLogo from './In-White-128px-TM.png';
import githubLogo from './GitHub-Mark-Light-120px-plus.png';
import arrow from './59691.png';
import './Header.css';



export default class Header extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      hide: true,
      arrow: false,
      hello: ""
    }
    this.helloWorld = "Hello World";
    this.hideTitle = this.hideTitle.bind(this);
    this.arrowFlash = this.arrowFlash.bind(this);
    this.writeHello = this.writeHello.bind(this);
  }
  componentDidMount() {
    window.addEventListener('scroll', this.hideTitle);
    setTimeout(() => this.setState({ hide: false }), 1000);
    this.writeHello();
    setTimeout(() => this.interval = setInterval(() => this.arrowFlash(), 1200), 1500);
  }
  componentWillUnmount() {
    window.removeEventListener('scroll', this.hideTitle);
    clearInterval(this.interval);
  }

  hideTitle() {
    window.scrollY > 0 ?
      !this.state.hide ? this.setState({ hide: true }) : null:
      this.state.hide ? this.setState({ hide: false }) : null;
    if (this.header.getBoundingClientRect().bottom <= 20 && this.state.hello === "Hello World") this.setState({ hello: "Hello Again" });
  }

  arrowFlash() {
    this.setState({ arrow: !this.state.arrow });
  }

  writeHello() {
    setTimeout(() => {
      const int = setInterval(() => {
        this.setState({hello: `${this.state.hello}${this.helloWorld.slice(0, 1)}`});
        this.helloWorld = this.helloWorld.slice(1);
        if (this.helloWorld.length === 0) clearInterval(int);
      }, 100);
    }, 1500);
  }
  
  render() {
    return (
      <div className="header" ref={(div) => this.header = div} >
        <Particles params={{
                                particles: {
                                    number: {
                                        value: 80
                                    },
                                    line_linked: {
                                        shadow: {
                                            enable: false,
                                            color: "#3CA9D1",
                                            blur: 5
                                        }
                                    }
                                },
                                interactivity: {
                                  detect_on: "window",
                                  events: {
                                    onhover: {
                                      enable: true,
                                      mode: "repulse"
                                    }
                                  },
                                  modes: {
                                    repulse: {
                                      distance: 100,
                                      duration: 0.4
                                    }
                                  }
                                }
                            }}
              style={{
                                position: "fixed",
                                top: 0,
                                left: 0,
                                width: "100%",
                                height: "100%"
                            }}/>  
        <div className="header-titles">
          <h1 className={this.state.hide ? "header-title-none" : "header-title"}>Full Stack</h1>
          <h1 className={this.state.hide ? "header-title-none" : "header-title"}>Software Engineer/Web Developer</h1>
          <h2 className="header-hello">{this.state.hello}</h2>
        </div>
        <img className={this.state.arrow ? "arrow" : "arrow-none"} src={arrow} />
        <div className="header-links">
          <a href="https://www.linkedin.com/in/dylanscheidt">
            <img className="header-icon" src={linkedinLogo} />
          </a>
          <a href="https://github.com/dys2" >
            <img className="header-icon" src={githubLogo} />
          </a>
        </div>
      </div>
    )
  }

}
