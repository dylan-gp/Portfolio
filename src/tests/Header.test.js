import React from 'react';
import Header from '../components/Header';
import {shallow, configure} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

configure({ adapter: new Adapter() });

test('Component should match the snapshot', () => {
  const component = renderer.create(<Header />);
  const json = component.toJSON();
  expect(json).toMatchSnapshot();
});

const header = shallow(<Header />);

test('Component should display proper header', () => {
  header
    .find('h1')
    .forEach((h1, i) => 
      i ?
        expect(h1.text()).toEqual('Software Engineer/Web Developer'):
        expect(h1.text()).toEqual('Full Stack')
    );
});

test('Component mounts with correct state', () => {
  const state = header.state();
  const expectedState = { 
    hide: true,
    arrow: false,
    hello: '',
    particle: true
  };
  expect(state).toMatchObject(expectedState);
});