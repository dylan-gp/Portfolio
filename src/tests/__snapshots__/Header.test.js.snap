// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component should match the snapshot 1`] = `
<div
  className="header"
>
  <div
    className={undefined}
  >
    <canvas
      className={undefined}
      style={
        Object {
          "height": "100%",
          "left": 0,
          "position": "fixed",
          "top": 0,
          "width": "100%",
        }
      }
    />
  </div>
  <div
    className="header-titles"
  >
    <h1
      className="header-title-none"
    >
      Full Stack
    </h1>
    <h1
      className="header-title-none"
    >
      Software Engineer/Web Developer
    </h1>
    <h2
      className="header-hello"
    >
      
    </h2>
  </div>
  <img
    alt="down arrow"
    className="arrow-none"
    src="59691.png"
  />
  <div
    className="header-links"
  >
    <a
      href="https://www.linkedin.com/in/dylanscheidt"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="linkedin"
        className="header-icon"
        src="In-White-128px-TM.png"
      />
    </a>
    <a
      href="https://github.com/dys2"
      rel="noopener noreferrer"
      target="_blank"
    >
      <img
        alt="github"
        className="header-icon"
        src="GitHub-Mark-Light-120px-plus.png"
      />
    </a>
  </div>
</div>
`;
